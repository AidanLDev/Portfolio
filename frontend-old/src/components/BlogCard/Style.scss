@import "pallet";
@import "variables";
@import "mixins";

.blogLinks {
  text-decoration: none;
  color: $black;
}

.blogCardContainer {
  border: 1px solid #888;
  background-color: #f7f7f7;
  box-shadow: 2px 2px 2px 2px #888;
  width: 80%;
  height: 200px;
  margin: 25px auto;
  padding: 5px 30px;
  // cursor: pointer;
}
