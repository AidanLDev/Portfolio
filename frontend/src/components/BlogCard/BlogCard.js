import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";

import styles from "./Style.scss";

const blogCard = props => {
  return (
    <NavLink to={"/blog/" + props.route} className={styles.blogLinks}>
      <div className={styles.blogCardContainer} onClick={props.clicked}>
        <h1 className={styles.blogHeading}>{props.title}</h1>
        <p>{props.description}</p>
      </div>
    </NavLink>
  );
};

blogCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  clicked: PropTypes.func,
  route: PropTypes.string
};

export default blogCard;
