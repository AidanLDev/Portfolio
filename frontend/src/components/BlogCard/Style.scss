.blogCardContainer {
  border: 1px solid #888;
  width: 80%;
  height: 200px;
  margin: auto;
  padding: 4px 0;
}
