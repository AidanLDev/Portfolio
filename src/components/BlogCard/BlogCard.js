import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";

import styles from "./Style.module.scss";

const blogCard = props => {
  return (
    <NavLink to={"/blog/" + props.route} className={styles.blogLinks}>
      <div className={styles.blogCardContainer} onClick={props.clicked}>
        <h1>{props.title}</h1>
        <p>{props.description}</p>
      </div>
    </NavLink>
  );
};

blogCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  clicked: PropTypes.func,
  route: PropTypes.string
};

export default blogCard;
