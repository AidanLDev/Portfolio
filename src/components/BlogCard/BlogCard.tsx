import React from 'react';
import { NavLink } from 'react-router-dom';

import styles from './Style.module.scss';

interface Props {
  clicked?:
    | ((event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void)
    | undefined;
  route: string;
  title: string;
  description: String;
}

const blogCard: React.FC<Props> = ({ clicked, route, title, description }) => {
  return (
    <NavLink to={'/blog/' + route} className={styles.blogLinks}>
      <div className={styles.blogCardContainer} onClick={clicked}>
        <h1>{title}</h1>
        <p>{description}</p>
      </div>
    </NavLink>
  );
};

export default blogCard;
