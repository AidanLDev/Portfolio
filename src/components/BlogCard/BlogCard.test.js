import React from 'react';
import { shallow } from 'enzyme';
import BlogCard from './BlogCard';

describe('<BlogCard />', () => {
  const component = shallow(
    <BlogCard title='About Aidans Life' description='once upon a time...' />
  );
  test('Renders BlogCard Component', () => {
    expect(component).toMatchSnapshot();
  });
});
