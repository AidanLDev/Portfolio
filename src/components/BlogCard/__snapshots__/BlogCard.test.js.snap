// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogCard /> Renders BlogCard Component 1`] = `
<NavLink
  className="blogLinks"
  to="/blog/undefined"
>
  <div
    className="blogCardContainer"
  >
    <h1>
      About Aidans Life
    </h1>
    <p>
      once upon a time...
    </p>
  </div>
</NavLink>
`;
